R/expect_null_linter.R

```diff
@@ -0,0 +1,48 @@
+#' expect_null Linter
+#'
+#' Require usage of expect_null(x) over expect_equal(x, NULL) and similar
+#'   [testthat::expect_null()] exists specifically for testing for `NULL` objects.
+#'   [testthat::expect_equal()], [testthat::expect_identical()], and
+#'   [testthat::expect_true()] can also be used for such tests,
+#'   but it is better to use the tailored function instead.
+#'
+#' @evalRd rd_tags("expect_null_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+expect_null_linter <- function() {
+  Linter(function(source_file) {
+    if (length(source_file$parsed_content) == 0L) {
+      return(list())
+    }
+
+    xml <- source_file$xml_parsed_content
+
+    # for expect_{equal,identical}(x, NULL)
+    equal_expr_cond <- sprintf(
+      "SYMBOL_FUNCTION_CALL[%s] and following-sibling::expr[2][NULL_CONST]",
```

Comment:
What about 'expect_equal(NULL, x)`?
Is there any downside to dropping the `[2]` component?
The `following-sibling::expr[NULL_CONST]` would require one argument (first or second) to be identical to the expression `NULL` I think.

Reply:
I think my concern here was avoiding thinking about other arguments to `expect_equal/expect_identical`, e.g. `expect_identical(x, y, tolerance=NULL)`. I haven't thought through whether it makes sense for such cases to exist but restricting to `postion() = 1 or position() = 2` should be fine.

Reply:
covered `expect_equal(NULL, x)` now